packages/styled/types/base.d.ts

```diff
@@ -69,11 +75,17 @@ export interface CreateStyledComponent<
    */
   <AdditionalProps extends {} = {}>(
     ...styles: Array<
-      Interpolation<
-        ComponentProps &
-          SpecificComponentProps &
-          AdditionalProps & { theme: Theme }
-      >
+      // tslint:disable-next-line no-null-undefined-union
+      | StyledOptions<
```

Comment:
the same thing should be reverted from here as well

Reply:
Maybe, but it affects the case [you mentioned above](https://github.com/emotion-js/emotion/pull/2333#discussion_r609606867). 
```js
const test2 = styled.div({
  color: 'red',
}, options) // ← This point
```
Now I think it was just a typo because `styled.div(...args)`  does not allow to pass `StyledOptions` as arg:

```js
export type CreateStyledComponent = <Props>(
  ...args: Interpolations
) => StyledComponent<Props>

export type CreateStyled = {
  <Props>(
    tag: StyledElementType<Props>,
    options?: StyledOptions
  ): (...args: Interpolations) => StyledComponent<Props>,
  [key: string]: CreateStyledComponent,
  bind: () => CreateStyled
}
```
So I'm going to revert related changes and remove the corresponding test.